src/UCTNode.cpp

```diff
@@ -61,22 +57,22 @@ bool UCTNode::create_children(Network & network,
         return false;
     }
     // acquire the lock
-    LOCK(get_mutex(), lock);
+    bool success = acquire_expanding();
+    if (!success) {
+        return false;
+    }
+
     // no successors in final state
     if (state.get_passes() >= 2) {
+        expand_cancel();
         return false;
     }
+
     // check whether somebody beat us to it (after taking the lock)
```

Comment:
Is this case still possible now? If we acquired_expanding then we are the only ones here, right?

Can we still re-expand nodes that were created under limited memory?

Reply:
This can probably be fixed by not bailing out if acquire_expanding was already set to EXPANDED?